src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java

```diff
@@ -352,6 +445,37 @@ private void lazyLoadNextAlignmentContext() {
     }
 
     /**
+     * Fix the quality of two elements that comes from an overlaping pair
```

Comment:
Sure, I will add it. Basically is the same algorithm as samtools is using, and I can provide a test where `CheckPileup` complains with overlapping reads because of the overlapping read-pairs...